timedctl/timedctl.py

```diff
@@ -25,708 +24,568 @@
-                        ":",
-                    )
-                    old_duration = datetime.timedelta(
-                        hours=int(hours),
-                        minutes=int(minutes),
-                        seconds=int(seconds),
-                    )
-                    # calculate the new duration
-                    report["attributes"]["duration"] = old_duration + duration
-                    # update report
-                    timed.reports.patch(
-                        report["id"],
-                        report["attributes"],
-                        {"task": task},
-                    )
-                else:
-                    # create report
-                    res = timed.reports.post(
-                        {
-                            "duration": duration,
-                            "comment": attr["comment"],
-                        },
-                        {"task": task},
+
+        if res.status_code == requests.codes["created"]:
+            msg(f"Activity {comment} started successfully.")
+            return
+        # handle exception
+        error_handler("ERR_ACTIVITY_START_FAILED")
+
+    def stop_activity(self):
+        """Stop current activity."""
+        if not self.timed.activities.current:
+            error_handler("ERR_NO_CURRENT_ACTIVITY")
+        else:
+            self.timed.activities.stop()
+        msg("Activity stopped successfully.")
+
+    def show_activity(self, short):
+        """Show current activity."""
+        current_activity = self.timed.activities.get(
+            filters={"active": True},
+            include="task,task.project,task.project.customer",
+        )
+        if current_activity:
+            activity_obj = current_activity[0]
+            comment = " > " + activity_obj["attributes"]["comment"] if not short else ""
+            start = activity_obj["attributes"]["from-time"].strftime("%H:%M:%S")
+            msg(
+                f"Current activity: {self.format_activity(activity_obj)}{comment}"+
+                f"  (Since {start})",
+            )
+        else:
+            error_handler("ERR_NO_CURRENT_ACTIVITY")
+
+    def restart_activity(self, date):
+        """Restart an activity."""
+        # stop current activity first
+        if self.timed.activities.current:
+            self.timed.activities.stop()
+            msg("Stopped current activity.")
+        # select an activity
+        activity_obj = self.select_activity(date)
+        # grab attributes
+        comment = activity_obj[1]
+        task_id = activity_obj[3]
+        res = self.timed.activities.start(
+            attributes={"comment": comment},
+            relationships={"task": task_id},
+        )
+        if res.status_code == requests.codes["created"]:
+            msg(f'Activity "{comment}" restarted successfully.')
+            return
+        # handle exception
+        error_handler("ERR_ACTIVITY_START_FAILED")
```

Comment:
this can be refactored to be more readable:
```suggestion
        if res.status_code != requests.codes["created"]:
              # handle exception
              error_handler("ERR_ACTIVITY_START_FAILED")
        msg(f'Activity "{comment}" restarted successfully.')
```